src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/AzureDevOpsTimelineTimeoutTests.cs

```diff
@@ -0,0 +1,106 @@
+using FluentAssertions;
+using Microsoft.DotNet.Internal.AzureDevOps;
+using Microsoft.DotNet.Internal.Testing.DependencyInjection.Abstractions;
+using Microsoft.DotNet.Internal.Testing.Utility;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.AzureDevOpsTimeline.Tests
+{
+    public partial class AzureDevOpsTimelineTimeoutTests
+    {
+        public static bool OperationCancelled = false;
+
+        [TestDependencyInjectionSetup]
+        public static class TestDataConfiguration
+        {
+            public static void Dependencies(IServiceCollection collection)
+            {
+
+                collection.AddOptions();
+                collection.AddLogging(logging =>
+                {
+                    logging.AddProvider(new NUnitLogger());
+                });
+                collection.Configure<AzureDevOpsTimelineOptions>((o, p) =>
+                {
+                    o.LogScrapingTimeout = "00:00:100";
+                });
+            }
+
+            public static Func<IServiceProvider, AzureDevOpsTimeline> Controller(IServiceCollection collection)
+            {
+                collection.AddScoped<AzureDevOpsTimeline>();
+                return s => s.GetRequiredService<AzureDevOpsTimeline>();
+            }
+
+            [ConfigureAllParameters]
+            public static Func<IServiceProvider, InMemoryTimelineTelemetryRepository> Repository(
+                IServiceCollection collection, string project, DateTimeOffset latestTime)
+            {
+                collection.AddSingleton<ITimelineTelemetryRepository>(
+                    s => new InMemoryTimelineTelemetryRepository(
+                        new List<(string project, DateTimeOffset latestTime)> { (project, latestTime) }
+                    )
+                );
+
+                return s => (InMemoryTimelineTelemetryRepository)s.GetRequiredService<ITimelineTelemetryRepository>();
+            }
+
+            public static void Clock(IServiceCollection collection, DateTimeOffset staticClock)
+            {
+                Mock<ISystemClock> mockSystemClock = new Mock<ISystemClock>();
```

Comment:
We already have a "TestClock", you can just use that one rather than having to mock it yourself. :-)

Reply:
I just copied the setup from the other test class, `AzureDevOpsTimelineTests`

Reply:
Replaced the `Mock<ISystemClock>` with the `TestClock` in both test classes